### PR TITLE
Manual merge of main and develop-3 

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # (issue)

I made an attempt at a merge of develop-3 with main. I was and am a bit confused as to the state of things so I took my own pass at manually merging the two branches together and visually checking the code differences between the two. Most of the changes were related to `master` vs `main`, plus I found a section on the Logger. I kept it in case it wasn't replicated elsewhere.

A comparison against the other latest merge candidate shows this pull has a few changes that are needed over there.

## Type of change

Please delete options that are not relevant.

- Merge issues

# How Has This Been Tested?
TravisCI
# Checklist:

- [x] Make sure you are merging into the ``v3-0-rc1`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/instruments/templates/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
```
Comment (rstoneback) [inline pysat/instruments/templates/__init__.py:1]: Unsure where this init popped up.

File: README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 </div>
 
 # pysat: Python Satellite Data Analysis Toolkit
-[![Build Status](https://travis-ci.com/pysat/pysat.svg?branch=master)](https://travis-ci.com/pysat/pysat)
+[![Build Status](https://travis-ci.org/pysat/pysat.svg?branch=main)](https://travis-ci.org/pysat/pysat)
```
Comment (rstoneback) [inline README.md:7]:
```suggestion
[![Build Status](https://travis-ci.com/pysat/pysat.svg?branch=main)](https://travis-ci.com/pysat/pysat)
```

Comment (rstoneback) [reply README.md:7]: I'm not sure why this is still showing a difference between main and master here. I think the diff is a bit out of date. Perhaps the tests on the new commits in the other branch need to run first?

**Comment by rstoneback** [comment]
@aburrell this could use your perusal. Tests and all that need to cycle but I'm preemptively raising the flag as it is relevant to the RC release.

File: pysat/tests/test_files.py [hunk 3]
```diff
@@ -934,70 +933,6 @@ def test_race_condition(self):
         proc_pool.map(create_instrument, range(processes))
 
 
-class TestDeprecation():
```
Comment (aburrell) [inline pysat/tests/test_files.py:937]: Would be a good time to put this in a develop PR for the 2.3.0 release.

**Review by aburrell — APPROVED** [review]
Looks good.  Comments are a matter of efficiency rather than issues with this PR.

**Pull request merged**